ginivision/src/main/java/net/gini/android/vision/internal/camera/photo/ExifReader.java

```diff
@@ -0,0 +1,63 @@
+package net.gini.android.vision.internal.camera.photo;
+
+import static org.apache.commons.imaging.Imaging.getMetadata;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import org.apache.commons.imaging.ImageReadException;
+import org.apache.commons.imaging.formats.jpeg.JpegImageMetadata;
+import org.apache.commons.imaging.formats.tiff.TiffField;
+import org.apache.commons.imaging.formats.tiff.constants.ExifTagConstants;
+
+import java.io.IOException;
+import java.util.Arrays;
+
+/**
+ * @exclude
+ */
+class ExifReader {
+
+    private final byte[] jpeg;
+
+    ExifReader(@NonNull final byte[] jpeg) {
+        this.jpeg = jpeg;
+    }
+
+    @NonNull
+    public String getUserComment() {
+        try {
+            final JpegImageMetadata jpegMetadata = (JpegImageMetadata) getMetadata(jpeg);
+            if (jpegMetadata == null) {
+                throw new ExifReaderException("No jpeg metadata found");
+            }
+
+            final TiffField userCommentField = jpegMetadata.findEXIFValue(
+                    ExifTagConstants.EXIF_TAG_USER_COMMENT);
+
+            try {
+                final byte[] rawUserComment = userCommentField.getByteArrayValue();
+                if (rawUserComment == null) {
+                    throw new ExifReaderException("No User Comment found");
+                }
+                return new String(Arrays.copyOfRange(rawUserComment, 8, rawUserComment.length));
+            } catch (ClassCastException e) {
+                throw new ExifReaderException("Unexpected type in User Comment: " + e.getMessage(), e);
+            }
+        } catch (IOException | ImageReadException e) {
+            throw new ExifReaderException("Could not read jpeg metadata: " + e.getMessage(), e);
+        }
+    }
+
+    @Nullable
+    String getValueForKeyfromUserComment(@NonNull final String key, @NonNull final String userComment) {
+        final String[] keyValuePairs = userComment.split(",");
+        for (final String keyValuePair : keyValuePairs) {
+            final String[] keyAndValue = keyValuePair.split("=");
+            if (keyAndValue[0].equals(key)) {
```

Comment:
Is referencing the 0 and 1 index directly safe here? I know we usually enforce a specific format for the user comment, but it's still an external EXIF field. And if users of out library choose to feed us custom images or add something themselves, an exception would be thrown.